.github/workflows/macos.yml

```diff
@@ -86,8 +86,10 @@ jobs:
       run: |
         pip install ${{ matrix.numpy-version }}
         pip install setuptools wheel cython pytest pytest-xdist pybind11 pytest-xdist mpmath gmpy2
+        pip install pythran
 
     - name: Test SciPy
       run: |
         export DYLD_LIBRARY_PATH=/usr/local/Cellar/gcc@6/6.5.0_5/lib/gcc/6/:$DYLD_LIBRARY_PATH
-        python -u runtests.py
+        SCIPY_USE_PYTHRAN=0 python -u runtests.py
+        SCIPY_USE_PYTHRAN=1 python -u runtests.py
```

Comment:
This runs the test suite twice, we don't want that. Let's try and set `SCIPY_USE_PYTHRAN` to `1` for py37/38, and to `0` for py39.